metaflow/plugins/kfp/kfp.py

```diff
@@ -28,35 +29,166 @@ def get_ordered_steps(graph):
     return ordered_steps
 
 
-def step_container_op(step_name, code_url=DEFAULT_FLOW_CODE_URL):
+def step_op_func(step_name: str,
+                 code_url: str,
+                 ds_root: str,
+                 run_id: str,
+                 task_id: str,
+                 prev_step_name: str,
+                 prev_task_id: str) -> NamedTuple('StepOutput', [('ds_root', str),
+                                                         ('run_id', str),
+                                                         ('next_step', str),
+                                                         ('next_task_id', str),
+                                                         ('current_step', str),
+                                                         ('current_task_id', str)]
+                                                      ):
     """
-    Method to create a kfp container op that corresponds to a single step in the flow.
+    Function used to create a KFP container op (see: `step_container_op`) that corresponds to a single step in the flow.
 
-    TODO: This does not maintain state. The custom pre-start command used below would be removed once we have state accessible across KFP steps.
-    TODO: The public docker is a copy of the internal docker image we were using (borrowed from aip-kfp-example). Check if any stage here may need to be further modified later.
     """
+    import subprocess
+    from collections import namedtuple
+
+    print("\n----------RUNNING: CODE DOWNLOAD from URL---------")
+    subprocess.call(["curl -o helloworld.py {}".format(code_url)], shell=True)
```

Comment:
KFP Lightweight components does not seem to allow me to move these common statements into a function and re-use it across both `step-op-func` and `pre-start-op-func`. Any suggestions on how to refactor this?

Reply:
Why isn't it allowing you to move them to a common function?

Reply:
add param `use_code_pickling=True` to `func_to_container_op`

Reply:
This should work but unfortunately, as per docs, it "is sensitive to version mismatch between python in component creation environment and runtime image". And, in trying to do this, I get an error due to python version mismatch between my creation environment (3.8.3) and the version on the docker image (3.6.8). 

Looks like the better way is to just leave the code as is instead of using the code_pickling feature because it isn't very reliable.